infra/build_image_from_state.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module to build a image from a specific commit, branch or pull request
+
+This module is allows each of the OSS Fuzz projects fuzzers to be built
+from a specific point in time. This feature can be used for implementations
+like continuious integration fuzzing and bisection to find errors
+"""
+import os
+import re
+import sys
+import subprocess
+
+from helper import build_image_impl
+from helper import check_project_exists
+from helper import docker_run
+from helper import env_to_docker_args
+from helper import get_absolute_path
+from helper import get_command_string
+from helper import get_dockerfile_path
+from helper import get_output_dir
+from helper import get_work_dir
+from helper import workdir_from_dockerfile
+from RepoManager import RepoManager
+
+
+def build_fuzzer_from_commit(project_name, commit, fuzzer_name, local_store_path,
+                              engine='libfuzzer',
+                              sanitizer='address',
+                              architecture='x86_64'):
+  """Builds a ossfuzz fuzzer at a  specific commit SHA.
+
+  Args:
+    project_name: The oss fuzz project name
+    commit: The commit SHA to build the fuzzers at
+    fuzzer_name: The name of the fuzzer that needs to be built
+    local_store_path: The full file path of a place where a temp git repo is stored
+    engine: The fuzzing engine to be used
+    sanitizer: The fuzzing sanitizer to be used
+    architecture: The system architiecture to be used for fuzzing
+
+  Returns:
+    0 on successful build 1 on failure
+  """
+  guessed_url = infer_main_repo(project_name, local_store_path, commit)
+  repo_man = RepoManager(guessed_url, local_store_path)
+  repo_man.checkout_commit(commit)
+  build_image_impl(project_name)
+  project_out_dir = get_output_dir(project_name)
+
+  # Clean old and possibly conflicting artifacts in project's out directory.
```

Comment:
rather than copy and pasting code from helper.py, we should refactor the relevant function (build_fuzzers) to support what we're doing here :) 

In fact, that function already has the relevant code needed: https://github.com/google/oss-fuzz/blob/df5a80bf9707e9e0b8f55a2f2b1df5ab9dd62cfe/infra/helper.py#L460

What we can do is just to refactor `build_fuzzers` into a `build_fuzzers_impl` or something similar which accepts the individual `arg` arguments like so:

```python
def build_fuzzers_impl(engine, sanitizer, project_name, source_path, .... etc):
  ...

def build_fuzzers(args):
  return build_fuzzers_impl(args.engine, args.sanitizer, etc...)
```

and use that here too.

Reply:
This makes more sense then what I was doing.